src/test/java/com/google/devtools/build/lib/analysis/actions/StrippingPathMapperTest.java

```diff
@@ -0,0 +1,186 @@
+package com.google.devtools.build.lib.analysis.actions;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.devtools.build.lib.actions.Artifact;
+import com.google.devtools.build.lib.actions.Spawn;
+import com.google.devtools.build.lib.analysis.ConfiguredTarget;
+import com.google.devtools.build.lib.analysis.FileProvider;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.rules.java.JavaCompilationArgsProvider;
+import com.google.devtools.build.lib.rules.java.JavaInfo;
+import java.io.IOException;
+import net.starlark.java.eval.Dict;
+import net.starlark.java.eval.Starlark;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link StrippingPathMapper}. */
+@RunWith(JUnit4.class)
+public class StrippingPathMapperTest extends BuildViewTestCase {
```

Comment:
@fmeum these tests don't work out very well internally. The paths are different `blaze-out` vs `bazel-out`. Instead of asserting for the exact paths, can we do regex matching instead? 

I also notice the order could be different as well. For example:

```
missing (6)
#1      : bazel-out/bin/java/com/google/test/B.java
#2      : bazel-out/bin/java/com/google/test/C.java
#3      : bazel-out/bin/java/com/google/test/liba-hjar.jar
#4      : bazel-out/bin/java/com/google/test/liba-hjar.jdeps
#5      : -XepOpt:foo:bar=bazel-out/bin/java/com/google/test/B.java
#6      : -XepOpt:baz=bazel-out/bin/java/com/google/test/C.java,bazel-out/bin/java/com/google/test/B.java

unexpected (6)
#1      : blaze-out/bin/java/com/google/test/liba-hjar.jar
#2      : blaze-out/bin/java/com/google/test/liba-hjar.jdeps
#3      : blaze-out/bin/java/com/google/test/B.java
#4      : blaze-out/bin/java/com/google/test/C.java
#5      : -XepOpt:foo:bar=blaze-out/bin/java/com/google/test/B.java
#6      : -XepOpt:baz=blaze-out/bin/java/com/google/test/C.java,blaze-out/bin/java/com/google/test/B.java
```

Reply:
I'm not a big fan of regexes here as they may make our test assertions less precise. I am instead using the product name now.

I am not asserting on the order in the Java test, only in the Starlark test where it should be identical between Bazel and Blaze.